data-collection/src/test/java/integration/IntegrationBaseTest.java

```diff
@@ -255,6 +255,26 @@ public void checkCKMethodNotFound() throws FileNotFoundException {
 		Assert.assertFalse(refactoringml.util.FileUtils.readFile("./logs/data-collection_test_ERROR.log").contains("CK did not find the refactored method:"));
 	}
 
+	@Test
+	public void monitorDuplicateStableInstances(){
+		this.session = sf.openSession();
+		int uniqueStableCommits = session.createSQLQuery("Select DISTINCT s.className, s.filePath, s.isTest, s.level, s.commitThreshold, s.classMetrics_id, s.commitMetaData_id, s.fieldMetrics_id, s.methodMetrics_id, s.processMetrics_id, s.project_id, s.variableMetrics_id From StableCommit s where s.project_id = " + project.getId())
```

Comment:
SQL trick: you can do a `select count(distinct ...) from table`, so that you don't have to send a huge table from the db to the program so that you do a `list.size` later.

Reply:
Then, something like `((Number) query.uniqueResult()).intValue()` to get a single integer directly from the query.

Reply:
great, I will do that

Reply:
Done

Reply:
I would be great to find a solution for this because we also use similar code in production:
https://github.com/refactoring-ai/predicting-refactoring-ml/blob/d0150fb1a0d4b6408dae1814a4b677d9c1596211/data-collection/src/main/java/refactoringml/db/Database.java#L96-L109

Reply:
Does it work in the production code, where the SQL is smaller?

Reply:
The nested select works, but is not as efficient as your suggested solution.

Reply:
> Does it work in the production code, where the SQL is smaller?

I will test the other solution in a few minutes

Reply:
So, right now, it works for the small query in `Database.java`, but doesn't work in the `IntegrationBaseTest`?

Reply:
Yes

Reply:
BTW, why do you need this `shortSession` in that method? Given that we open the session at the beginning of the main loops, aren't we always with an open session? (We should, as much as possible, push all transaction and connection-related code to the main loops, so that they are centralised in a single place).

Reply:
For the integration test, maybe too many columns there... It should work. But I wouldn't spend a lot of time in the test code, as it's ok not to be highly performant there. Just rollback to the way you were doing before!

Reply:
> BTW, why do you need this `shortSession` in that method? Given that we open the session at the beginning of the main loops, aren't we always with an open session? (We should, as much as possible, push all transaction and connection-related code to the main loops, so that they are centralised in a single place).

Yes, because it is called after the main loop.
I will check for other similar cases, we might be able to remove. I will do this on another branch.

Reply:
> For the integration test, maybe too many columns there... It should work. But I wouldn't spend a lot of time in the test code, as it's ok not to be highly performant there. Just rollback to the way you were doing before!

Ok, then this PR is done.